File: hw1/student/student.py
class Student:
    def __init__(self,first:str,last:str,grade:float)->None:
        self._first=first
        self._last=last
        self._grade:float=grade

    @property
    def fullname(self)->str:
        return '{} {}'.format(self._first,self._last)

    @property
    def email(self)->str:
         return '{}.{}@email.com'.format(self._first,self._last)
    
    def curvegrade(self)->float:
        return self._grade+1.0

File: hw2/src/student.cpp
#include "Student.h"

using namespace std;

namespace student{
    Student::Student(string first,string last,double grade){
        m_first=first;
        m_last=last;
        m_grade=grade;
    }
    string Student::fullname(){
        return m_first+" "+ m_last;
    }

    string Student::email(){
        return m_first+"."+ m_last+"@email.com";
    }
    double Student::curvegrade(){
        return m_grade+1.0;
    }
    string Student::getfirst(){
        return m_first;
    }
    string Student::getlast(){
        return m_last;
    }
    double Student::getgrade(){
        return m_grade;
    }
}